lib/inventory/controller.go

```diff
@@ -288,11 +320,15 @@ func (c *Controller) RegisterControlStream(stream client.UpstreamInventoryContro
 	// as much as possible. this is intended to mitigate load spikes on auth restart, and is reasonably
 	// safe to do since the instance resource is not directly relied upon for use of any particular teleport
 	// service.
-	ticker := interval.NewMulti(interval.SubInterval[intervalKey]{
+	ticker := interval.NewMulti(c.clock, interval.SubInterval[intervalKey]{
 		Key:              instanceHeartbeatKey,
 		VariableDuration: c.instanceHBVariableDuration,
 		FirstDuration:    fullJitter(c.instanceHBVariableDuration.Duration()),
 		Jitter:           seventhJitter,
+	}, interval.SubInterval[intervalKey]{
+		Key:      instanceTimeReconciliation,
+		Duration: c.timeReconciliationInterval,
+		Jitter:   seventhJitter,
```

Comment:
That's not really going to guarantee that the first ping is done by the time we do the first instance heartbeat keepalive - IMO waiting 3 more minutes is not really a problem, since we have to handle outdated agents not sending us the local time anyway.

Reply:
Well, it will initiate the ping but we have no guarantee that we'll actually receive the response before the heartbeat hits, especially if we rolled very low on `firstDuration`. This is why we should make sure that we won't report anything about a server if we haven't received a timestamp yet.

Reply:
ah, you mean that response of the ping must be after heart, we should set `LastMeasurement = nil` in such case https://github.com/gravitational/teleport/pull/44489/commits/ebbbde65c81cb7e3de69ac19da7d57aa9cdfbf2d
so it must be ignored in general check